Adds 1.9.0.1 prerelease as a v2 option.

telser opened the pull request:
Happened to notice that the first release candidate for stack-1.9 has been made [available](https://github.com/commercialhaskell/stack/releases/tag/v1.9.0.1). 

In particular they're asking for testing with the static bindist as they plan on making that the new default with 1.9. There are what seem like quite a few enhancements/fixes at the link, even discounting the Windows ones (that are mentioned first, not sure why) which won't be an impact on anything running Linux.

telser committed:
Adds 1.9.0.1 prerelease as a v2 option.

qxjit commented:
@telser Do you think we should include `-prerelease` in the docker tag? If so, I'll rename the directory to match (both for clarity, and because the `build` script requires it)

telser commented:
@qxjit I think that is probably a good idea. But I also think that the version of 1.9.0.1 is meant to only be used as a prerelease version number with 1.9.1 being the "release". Being more explicit is good IMO.

qxjit committed:
Renames 1.9.0.1 to 1.9.0.1-prerelease

qxjit committed:
Added --cache-from flipstone/stack to build script

This allows `docker build` to flipstone/stack images you have pulled
from remote (rather than building locally) to be used a source for
layer caching when building new images.

qxjit committed:
Adds 1.9.0.1-prerelease to the README list
